.github/workflows/generate-zip.yml

```
@@ -0,0 +1,64 @@
+name: Generate ZIP file
+
+on: [pull_request]
+
+jobs:
+  generate-zip-file:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Cache node_modules
+        id: cache-node-modules
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Setup node version and npm cache
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+
+      - name: Install Node dependencies
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: npm ci --no-optional
+
+      - name: Generate ZIP file
+        run: npm run package-plugin:deploy
+
+      - name: Create temp folder
+        run: mkdir temp
+
+      - name: Rename and move ZIP file
+        run: mv woocommerce-gutenberg-products-block.zip temp/woocommerce-gutenberg-products-block-${{ github.event.pull_request.number }}.zip
+
+      - name: Transfer ZIP file via SFTP
+        uses: AbleLincoln/push-to-sftp@v1.0
+        with:
+          host: ${{ secrets.FTP_HOST }}
+          port: 22
+          username: ${{ secrets.FTP_USER }}
+          password: ${{ secrets.FTP_PASS }}
+          sourceDir: ./temp/
+          targetDir: ${{ secrets.FTP_DIR }}
+
+      - name: Add release ZIP URL as comment to the PR
+        uses: mshick/add-pr-comment@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.API_TOKEN_GITHUB }}
```

At the moment, it adds comments from your account. I think that we should add the comment using the [Rubik User Bot](https://github.com/rubikuserbot) account. We can use the same GitHub token that is used here: https://github.com/woocommerce/woocommerce-blocks/blob/trunk/.github/workflows/automate-team-review-assignment-config.yml#L15

```suggestion
          GITHUB_TOKEN: ${{ secrets.PAT_FOR_ACTIONS }}
```

---

Thanks for pointing this out, @tarunvijwani. I just committed your suggestion.